yarnitor/background/worker.py

```diff
@@ -34,260 +38,390 @@
+    def get_url(self, path, **params):
+        """Issues an HTTP GET against the given path on the app tracking server with
+        the given parameters and treats the response as JSON.
 
         Parameters
         ----------
-        url : string
-        params : dict
+        path: str
+            Path to append to the root YARN RM path
+        **params, dict
+            Query parameters to append to the YARN RM URL
 
         Returns
         -------
         dict
+            JSON decoded response
         """
-        final_url = (self.prefix + url).format(version=self.version, **self.__dict__)
-        # Under some security models for YARN going to the tracking url
-        # required clicking through a security prompt.
-        # This presets that cookie.
+        url = '{tracking_url}/{path}'.format(tracking_url=self.tracking_url, path=path)
+        # Under some security models, the YARN proxy requires that a user click a link in
+        # order to access the tracking URL. Setting a cookie has the same effect, programmatically.
         cookies = {"checked_{}".format(self.application_id): 'true'}
-        resp = requests.get(final_url, params, cookies=cookies, timeout=10)
+        resp = requests.get(url, params, cookies=cookies, timeout=10)
         return resp.json()
 
     def generate_standardized_info(self, yarn_application_info):
-        """Generates the standardized dictionary of fields that are expected by
-        listing applications.
+        """Transforms information from the YARN ResourceManager and the YARN ApplicationMaster
+        into a dictionary of standard fields used by the frontend.
 
         Parameters
         ----------
-        yarn_application_info
+        yarn_application_info: dict
+            Information about a single app retrieved using YARNHandler.cluster_applications()
 
         Returns
         -------
         dict
+            Fields to copy verbatim from the YARN information about the app plus an
+            empty 'progress' list to be populated by more specific subtypes
         """
         verbatim_fields = ["id", "name", "user", "applicationType", "queue",
                            "startedTime", "allocatedMB", "allocatedVCores",
                            "trackingUrl", "state", "memorySeconds",
                            "vcoreSeconds"]
-
         r = {k: yarn_application_info[k] for k in verbatim_fields}
-        # defaults
-        r["job"] = "1"
-
-        # Progress should contain various progress records formatted as
-        # follows.
-        # This is the fallback progress handler for when we do not have a
-        # handler for the specific yarn application
-        # type.
-        r["progress"] = [
-            {
-                "name": "yarn-progress",
-                "completed": int(yarn_application_info["progress"]),
-                "failed": 0,
-                "running": 0,
-                "total": 100
-            }
-        ]
```

Comment:
what purpose did this code path serve?

Reply:
None. Leftover and dead. The frontend is robust against no progress data. Before it would populate a incorrect "100 total" for unknown reasons.